akka-projection-core/src/main/scala/akka/projection/internal/HandlerLifecycleAdapter.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.internal
+
+import scala.concurrent.Future
+
+import scala.compat.java8.FutureConverters._
+import akka.Done
+import akka.annotation.InternalApi
+import akka.projection.javadsl
+import akka.projection.scaladsl
+
+/**
+ * INTERNAL API: Adapter from javadsl.HandlerLifecycle to scaladsl.HandlerLifecycle
+ */
+@InternalApi
+private[akka] class HandlerLifecycleAdapter(delegate: javadsl.HandlerLifecycle) extends scaladsl.HandlerLifecycle {
```

Comment:
This is an adapter for only the lifecycle part. We can't use the generic one because it is made for the default Handler which is not good for JDBC. 

Btw, the existing one must move to core and eventually extend this one. Object for another PR.